File: web-api/src/main/java/br/edu/ifpb/ads/ws/service/ClienteService.java
package br.edu.ifpb.ads.ws.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import br.edu.ifpb.ads.ws.model.Cliente;
import br.edu.ifpb.ads.ws.repository.ClienteRepository;

/**
 * 
 * @author franck
 *
 */
@Service
public class ClienteService {
	
	@Autowired
	private ClienteRepository clienteRepository;
	
	public Cliente save(Cliente cliente){
		return clienteRepository.save(cliente);
	}
	
	public List<Cliente> getAll(){
		return clienteRepository.findAll();
	}
	
	public Cliente getById(Long id){
		return clienteRepository.findOne(id);
	}
	
	public List<Cliente> findByNome(String nome){
		return clienteRepository.findByNomeStartingWithIgnoreCase(nome);
	}
	
	public void delete(Cliente cliente){
		clienteRepository.delete(cliente);
	}
	
	public Cliente update(Cliente cliente){
		return clienteRepository.save(cliente);
	}
}

File: src/main/resources/static/app/components/cliente/cliente.service.js
angular.module('appCliente')

.factory('resourceFactory', function($resource) {
	
	var _private = $resource('/admin/clientes/:clienteId', null, {
		update : {
			method: 'PUT'
		}
	});

	return {
		private: _private
	};
})

.service('clienteService', function(resourceFactory){
	
	this.saveCliente = function(cliente){
		return resourceFactory.private.save(cliente);
	};
	
	this.updateCliente = function(cliente) {
		return resourceFactory.private.update(cliente);
	};
	
	this.getById = function(clienteId){
		return resourceFactory.private.get({clienteId: clienteId});
	};
	
	this.findAll = function() {
		return resourceFactory.private.query();
	};
	
	this.remove = function(clienteId) {
		return resourceFactory.private.delete({clienteId: clienteId});
	};

});